lambdas/report-user-identity/src/main/java/uk/gov/di/ipv/core/reportuseridentity/ReportUserIdentityHandler.java

```diff
@@ -0,0 +1,196 @@
+        }
+    }
+
+    private void processUsersToFindLOCAndUpdateDb(List<String> userIds)
+            throws CredentialParseException, HttpResponseExceptionWithErrorBody, ParseException,
+                    BatchDeleteException {
+        LOGGER.info(
+                LogHelper.buildLogMessage(
+                        String.format("Checking identity for total (%s) users.", userIds.size())));
+        List<ReportUserIdentityItem> reportUserIdentityItems = new ArrayList<>();
+        for (String userId : userIds) {
+            var tacticalVcs = verifiableCredentialService.getVcs(userId);
+            if (!userIdentityService.areVcsCorrelated(tacticalVcs)) {
+                LOGGER.info(
+                        LogHelper.buildLogMessage(
+                                String.format("User (%s) VCs not correlated.", userId)));
+                continue;
+            }
+            boolean migrated = tacticalVcs.stream().allMatch(vc -> vc.getMigrated() != null);
+            var votAttained =
+                    reportUserIdentityService.getStrongestAttainedVotForCredentials(tacticalVcs);
+
+            reportUserIdentityItems.add(
+                    new ReportUserIdentityItem(
+                            userId,
+                            votAttained.orElse(Vot.P0).name(),
+                            reportUserIdentityService.getIdentityConstituent(tacticalVcs),
+                            migrated));
+        }
+        LOGGER.info(LogHelper.buildLogMessage("Updating processed user's identity."));
+        reportUserIdentityDataStore.createOrUpdate(reportUserIdentityItems);
+    }
+
+    private ReportProcessingResult buildReportProcessingResult() {
+        LOGGER.info(LogHelper.buildLogMessage("Building report processing summary result."));
+        List<ReportUserIdentityItem> totalP2Identities =
+                reportUserIdentityDataStore.getItems(ATTR_NAME_TO_SUMMARISE_ON, Vot.P2.name());
+        long totalP2 = totalP2Identities.size();
+        long totalP2Migrated =
+                totalP2Identities.stream()
+                        .filter(ReportUserIdentityItem::isMigrated)
+                        .toList()
+                        .size();
+        long totalPCL250 =
+                reportUserIdentityDataStore
+                        .getItems(ATTR_NAME_TO_SUMMARISE_ON, Vot.PCL250.name())
+                        .size();
+        long totalPCL200 =
+                reportUserIdentityDataStore
+                        .getItems(ATTR_NAME_TO_SUMMARISE_ON, Vot.PCL200.name())
+                        .size();
+        long totalP1 =
+                reportUserIdentityDataStore
+                        .getItems(ATTR_NAME_TO_SUMMARISE_ON, Vot.P1.name())
+                        .size();
+        long totalP0 =
+                reportUserIdentityDataStore
+                        .getItems(ATTR_NAME_TO_SUMMARISE_ON, Vot.P0.name())
+                        .size();
```

Comment:
As with the first scan, this is very inefficient - this is doing 5 full table scans which each return a complete list of records.

Instead, I think we need to:
- Perform a single table scan, returning all relevant details
- Aggregate each page before fetching the next

That way we can calculate all the stats in one scan, and without needing to hold the full dataset in memory.

Reply:
reduce it to one paginated call to gather all users identity for aggregation.